src/google/protobuf/compiler/java/helpers.cc

```diff
@@ -1120,6 +1122,22 @@ void EscapeUtf16ToString(uint16_t code, std::string* output) {
   }
 }
 
+void MaybeSplitJavaMethod(io::Printer *printer, int* fields_in_function, int* method_num,
```

Comment:
Why do we MaybeSplitJavaMethod? Can't we know ahead of time (by the number of fields) how we need to split the top-level method?